common/src/java/org/apache/hadoop/hive/common/StringInternUtils.java

```diff
@@ -135,10 +135,10 @@ public static Path internUriStringsInPath(Path path) {
 
   public static <K> Map<K, String> internValuesInMap(Map<K, String> map) {
     if (map != null) {
-      for (K key : map.keySet()) {
-        String value = map.get(key);
+      for (Map.Entry<K, String> entry : map.entrySet()) {
+        String value = entry.getValue();
         if (value != null) {
-          map.put(key, value.intern());
+          map.put(entry.getKey(), value.intern());
```

Comment:
I know it's probably out of scope in this patch - but in case the value is already "interned" - then it will not change; at a cost of a conditional we can skip the hash lookup(put) as well

Reply:
Nice idea! I followed similar logic to check if values are already interned in all helper methods in StringInternUtils class